### PR TITLE
Add table gcp_kms_key_version closes #342

### LINKED ISSUE #342: Add table gcp_kms_key_version
**References**
This table is required for GCP reporting.
https://pkg.go.dev/google.golang.org/api@v0.48.0/cloudkms/v1#ProjectsLocationsKeyRingsCryptoKeysService.CryptoKeyVersions

### PR DESCRIPTION
**Pull request by karanpopat**
# Integration test logs
<details>
  <summary>Logs</summary>
  
```
Running SQL query: test-get-query.sql
[
  {
    "crypto_key_version": 1,
    "name": "turbottest70248",
    "self_link": "https://cloudkms.googleapis.com/v1/projects/parker-aaa/locations/global/keyRings/turbottest70248/cryptoKeys/turbottest70248/cryptoKeyVersions/1"
  }
]
✔ PASSED

Running SQL query: test-list-query.sql
[
  {
    "akas": [
      "gcp://cloudkms.googleapis.com/projects/parker-aaa/locations/global/keyRings/turbottest70248/cryptoKeys/turbottest70248/cryptoKeyVersions/1"
    ],
    "crypto_key_version": 1,
    "name": "turbottest70248",
    "self_link": "https://cloudkms.googleapis.com/v1/projects/parker-aaa/locations/global/keyRings/turbottest70248/cryptoKeys/turbottest70248/cryptoKeyVersions/1"
  }
]
✔ PASSED

Running SQL query: test-not-found-query.sql
null
✔ PASSED

Running SQL query: test-turbot-query.sql
[
  {
    "crypto_key_version": 1,
    "name": "turbottest70248",
    "self_link": "https://cloudkms.googleapis.com/v1/projects/parker-aaa/locations/global/keyRings/turbottest70248/cryptoKeys/turbottest70248/cryptoKeyVersions/1"
  }
]
✔ PASSED

POSTTEST: tests/gcp_kms_key_version

TEARDOWN: tests/gcp_kms_key_version

SUMMARY:

1/1 passed.

```
</details>

# Example query results
<details>
  <summary>Results</summary>
  
```
> select
  name,
  crypto_key_version,
  state
from
  gcp_kms_key_version limit 3;
+----------------+--------------------+-----------+
| name           | crypto_key_version | state     |
+----------------+--------------------+-----------+
| rupkey-test-01 | 10                 | DESTROYED |
| rupkey-test-01 | 3                  | DESTROYED |
| rupkey-test-01 | 12                 | ENABLED   |
+----------------+--------------------+-----------+


> select name, crypto_key_version, self_link from gcp_kms_key_version where name = 'turbottest82190' and location = 'global' and key_ring_name = 'turbottest82190' and crypto_key_version = 1
+-----------------+--------------------+-------------------------------------------------------------------------------------------------------------------------------------------------+
| name            | crypto_key_version | self_link                                                                                                                                       |
+-----------------+--------------------+-------------------------------------------------------------------------------------------------------------------------------------------------+
| turbottest82190 | 1                  | https://cloudkms.googleapis.com/v1/projects/parker-aaa/locations/global/keyRings/turbottest82190/cryptoKeys/turbottest82190/cryptoKeyVersions/1 |
+-----------------+--------------------+-------------------------------------------------------------------------------------------------------------------------------------------------+

```
</details>

### CODE REVIEW

File: gcp/table_gcp_kms_key_version.go [hunk 1]
```diff
@@ -0,0 +1,315 @@
...
+		}
+		wg.Done()
+	}
+	wg.Wait()
+
+	return nil, nil
+}
+
+func listKeyDetailsForRings(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) ([]*cloudkms.CryptoKey, error) {
+
```
Comment (bigdatasourav) [inline gcp/table_gcp_kms_key_version.go:205]:
```suggestion
```

File: gcp/table_gcp_kms_key_version.go [hunk 2]
```diff
@@ -0,0 +1,315 @@
...
+	key_ring_name := strings.Split(key.Name, "/")[5]
+	resource_name := strings.Split(key.Name, "/")[7]
+	crypto_key_version := strings.Split(key.Name, "/")[9]
+
+	turbotData := map[string]interface{}{
+		"Project":           project,
+		"Location":          location,
+		"KeyRing":           key_ring_name,
+		"ResourceName":      resource_name,
+		"CryptoKeyVersions": crypto_key_version,
```
Comment (bigdatasourav) [inline gcp/table_gcp_kms_key_version.go:283]:
```suggestion
		"CryptoKeyVersion": crypto_key_version,
```

File: gcp/table_gcp_kms_key_version.go [hunk 3]
```diff
@@ -0,0 +1,315 @@
...
+	pageSize := types.Int64(1000)
+	limit := d.QueryContext.Limit
+	if d.QueryContext.Limit != nil {
+		if *limit < *pageSize {
+			pageSize = limit
+		}
+	}
+
+	// Split CryptoKeys in chunks of 50 for concurrent fetching of CryptoKeyVersions
+	chunksCryptoKeys := chunkSlice(cryptoKeys, 50)
```
Comment (bigdatasourav) [inline gcp/table_gcp_kms_key_version.go:173]: As discussed please update the list call to use async call in CryptoKeys.List API itself.

File: docs/tables/gcp_kms_key_version.md [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+  crypto_key_version,
+  state
+from
+  gcp_kms_key_version
+where
+  create_time <= (current_date - interval '30' day)
+order by
+  create_time;
+```
+
```
Comment (bigdatasourav) [inline docs/tables/gcp_kms_key_version.md:33]: Can we please add an example query for the `latest key version disabled`

File: gcp/table_gcp_kms_key_version.go [hunk 5]
```diff
@@ -0,0 +1,315 @@
...
+				Name:        "state",
+				Description: "The current state of the CryptoKeyVersion.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "algorithm",
+				Description: "The CryptoKeyVersionAlgorithm that this CryptoKeyVersion supports.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
```
Comment (bigdatasourav) [inline gcp/table_gcp_kms_key_version.go:67]: Columns with the type `JSON` should be at last in ascending order.

**Review by bigdatasourav — APPROVED** [review]
LGTM

File: docs/tables/gcp_kms_key_version.md [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+### Basic info
+
+```sql
+select
+  name,
+  crypto_key_version,
+  state
+from
+  gcp_kms_key_version;
```
Comment (rajlearner17) [inline docs/tables/gcp_kms_key_version.md:15]:
```suggestion
  gcp_kms_key_version
  where state <> 'DESTROYED'
```

**Pull request merged**